sssom/io.py

```diff
@@ -85,42 +77,41 @@ def validate_file(input_path: str):
         return False
 
 
-def split_file(input_path: str, output_directory: str):
+def split_file(input_path: str, output_directory: str) -> None:
     """
-    Splits an SSSOM TSV by prefixes and relations.
+    Split an SSSOM TSV by prefixes and relations.
 
     Args:
-        input_path (str): The path to the input file in one of the legal formats, eg obographs, aligmentapi-xml
-        output_directory (str): The directory to which the split file should be exported.
-
-    Returns:
-
+        input_path: The path to the input file in one of the legal formats, eg obographs, aligmentapi-xml
+        output_directory: The directory to which the split file should be exported.
     """
-    if validators.url(input_path) or os.path.exists(input_path):
-        msdf = read_sssom_table(input_path)
-        splitted = split_dataframe(msdf)
-        write_tables(splitted, output_directory)
-    else:
-        raise Exception(f"{input_path} is not a valid file path or url.")
+    raise_for_bad_path(input_path)
+    msdf = read_sssom_table(input_path)
+    splitted = split_dataframe(msdf)
+    write_tables(splitted, output_directory)
 
 
-def get_metadata_and_curie_map(metadata_path, curie_map_mode: str = "metadata_only"):
+def get_metadata_and_curie_map(
+    metadata_path: Optional[str] = None, curie_map_mode: str = "metadata_only"
```

Comment:
type annotations give a lot of context to these functions, especially since most are missing docs ATM

Reply:
will add extensive docs everywhere.

Reply:
Definitely on the todo.